pkg/controller/plan/adapter/ovirt/client.go

```diff
@@ -64,16 +65,44 @@ func (r *Client) RemoveSnapshots(vmRef ref.Ref, precopies []planapi.Precopy) (er
 		return
 	}
 	snapsService := vmService.SnapshotsService()
-	for i := range precopies {
-		snapService := snapsService.SnapshotService(precopies[i].Snapshot)
-		_, err = snapService.Remove().Async(true).Query("correlation_id", r.Migration.Name).Send()
-		if err != nil {
-			err = liberr.Wrap(err)
+	go func() {
+		for i := range precopies {
+			snapService := snapsService.SnapshotService(precopies[i].Snapshot)
+			correlationID := fmt.Sprintf("precopy_%d_finalize", i)
```

Comment:
yeah it's temporary, I had some issues with the snapshot id, I didn't look into it yet so I placed this correlation id for now

Reply:
Added the start of the snapshot's id, there is a limit on the correlation id length. I can just put the entire id and that's it, but I think it would make it harder to trace in the logs